tests/lib.rs

```diff
@@ -93,6 +94,82 @@ fn counter_trivial() {
     );
 }
 
+#[test]
+fn gas_consumtion_works() {
+    let counter = Counter::new(99);
+
+    let store = MS::new();
+
+    let code =
+        include_bytes!("../target/wasm32-unknown-unknown/release/counter.wasm");
+
+    let contract = Contract::new(counter, code.to_vec(), &store).unwrap();
+
+    let mut network = NetworkState::<MS>::default();
+
+    let contract_id = network.deploy(contract).expect("Deploy error");
+
+    let mut gas = GasMeter::with_limit(1_000_000_000);
+
+    assert_eq!(
+        network
+            .transact::<_, ()>(contract_id, counter::INCREMENT, &mut gas)
+            .expect("Transaction error"),
+        ()
+    );
+
+    assert_eq!(
+        network
+            .query::<_, i32>(contract_id, counter::READ_VALUE, &mut gas)
+            .expect("Query error"),
+        100
+    );
+    assert_ne!(gas.spent(), 0);
+    assert!(gas.gas_left() < 1_000_000_000);
+}
+
+#[test]
+fn out_of_gas_aborts_execution() {
+    let counter = Counter::new(99);
+
+    let store = MS::new();
+
+    let code =
+        include_bytes!("../target/wasm32-unknown-unknown/release/counter.wasm");
+
+    let contract = Contract::new(counter, code.to_vec(), &store).unwrap();
+
+    let mut network = NetworkState::<MS>::default();
+
+    let contract_id = network.deploy(contract).expect("Deploy error");
+
+    let mut gas = GasMeter::with_limit(1);
+
+    assert!(network
```

Comment:
I would actually check that this is the error you're expecting.
To do so, you can use a combination of `assert!` and `matches!` macros. See: https://doc.rust-lang.org/std/macro.matches.html